File: bit_manipulation.py
#Bit Manipulation
#n = number on which operations are performed
#i = position
#k = changing value number

def getbit(n, i):
    return 1 if ((1 << i) & n) != 0 else 0
def setbit(n, i):
    return (n | (1 << i))
def clearbit(n, i):
    return (n & ~(1 << i))
def togglebit(n, i):
    return (n ^ (1 << i))
def updatebit(n, i, k):
    mask = ~(1 << i)
    n = n & mask
    return (n | (k << i))


print(getbit(5,1))
print(setbit(5,1))
print(clearbit(5,1))
print(togglebit(5,1))
print(updatebit(5,1,1))

File: bit_manipulation.c
/*  Bit Manipulation
        n = number on which operations are performed
        i = position
        k = changing value number
*/

#include<stdio.h>



int getbit(int n, int i){
    if(((1 << i) & n ) != 0)
        return 1;
    else
        return 0;
}

int setbit(int n, int i){
    return (n | (1 << i));
}

int clearbit(int n, int i){
    return (n & ~(1 << i));
}

int togglebit(int n, int i){
    return (n ^ (1 << i));
}

int updatebit(int n, int i, int k){
    int mask = ~(1 << i);
    n = n & mask;
    return (n | (k << i));
}

void main(){
    
    printf("%d\n",getbit(5,1));
    printf("%d\n",setbit(5,1));
    printf("%d\n",clearbit(5,1));
    printf("%d\n",togglebit(5,1));
    printf("%d\n",updatebit(5,1,1));
}
